src/SkeletalTrapezoidation.cpp

```diff
@@ -1321,21 +1322,19 @@ bool SkeletalTrapezoidation::isEndOfCentral(const edge_t& edge_to) const
 
 void SkeletalTrapezoidation::generateExtraRibs()
 {
-    auto end_edge_it = --graph.edges.end(); // Don't check newly introduced edges
-    for (auto edge_it = graph.edges.begin(); std::prev(edge_it) != end_edge_it; ++edge_it)
+    for (auto edge : graph.edges | ranges::views::drop_last(1))
```

Comment:
No remarks about the updated ranges-style code, but I find the original code here really weird. As such I'm not so sure if the suggested changes are equivalent to the original code.

Let's try to rewrite the iterator to a super basic index-based for loop.

```cpp
auto end_edge_it = --graph.edges.end();
```
Is rewritten to
```cpp
auto end_edge_index = graph.edges.size() - 1;
```
And the for loop
```cpp
for (auto edge_it = graph.edges.begin(); std::prev(edge_it) != end_edge_it; ++edge_it)
```
Translates to
```cpp
for (auto edge_index = 0; edge_index - 1 != end_edge_index; ++edge_index)
```

Lets simplify the for loop a bit
```cpp
for (auto edge_index = 0; edge_index - 1 != graph.edges.size() - 1; ++edge_index)
for (auto edge_index = 0; edge_index != graph.edges.size(); ++edge_index)
for (auto edge_index = 0; edge_index < graph.edges.size(); ++edge_index)
```

All edges are visited according to this logic; evalualting the following code
```cpp
#include <iostream>
#include <vector>

int main() {
  std::vector<int> v = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
  auto end_edge_it = --v.end();
  for (auto edge_it = v.begin(); std::prev(edge_it) != end_edge_it; ++edge_it)
  {
    int value = *edge_it;
    std::cout << "value: " << value << std::endl;
  }
}
```
produces
```
value: 1
value: 2
value: 3
value: 4
value: 5
value: 6
value: 7
value: 8
value: 9
value: 10
```

Please verify my logic, but in my opinion the `drop_last(1)` should not be added if we don't want to change the result of the program. 

We can still argue about the intention and validity of the original code.

```suggestion
    for (auto edge : graph.edges)
```

Reply:
> Please verify my logic, but in my opinion the drop_last(1) should not be added if we don't want to change the result of the program.

Damn you and your logic, it holds!